py_ecc/optimized_bls12_381/optimized_curve.py

```diff
@@ -109,9 +116,9 @@ def add(p1, p2):
 
 
 # Elliptic curve point multiplication
-def multiply(pt, n):
+def multiply(pt: Optimized_Point3D[Optimized_Field], n: int) -> Optimized_Point3D[Optimized_Field]:
     if n == 0:
-        return (pt[0].__class__.one(), pt[0].__class__.one(), pt[0].__class__.zero())
+        return (type(pt[0]).one(), type(pt[0]).one(), type(pt[0]).zero())
```

Comment:
I *think* you can actually drop the `type` call entirely and just do `pt[0].one()` since class methods can be called on instances.

If this works there appear to be quite a few places this cleanup could be applied.

Reply:
In case this is about performance, a hacky local test to guide intuition:

```py
In [7]: def r(n=10000000):
   ...:     st = perf_counter()
   ...:     for _ in range(n):
   ...:         a.b
   ...:     print(perf_counter() - st)
   ...:     

In [11]: r()
0.3565350249991752

In [12]: def r2(n=10000000):
    ...:     st = perf_counter()
    ...:     for _ in range(n):
    ...:         A.b
    ...:     print(perf_counter() - st)
    ...:     

In [13]: r2()
0.3344530600006692

In [15]: def r3(n=10000000):
    ...:     st = perf_counter()
    ...:     for _ in range(n):
    ...:         type(a).b
    ...:     print(perf_counter() - st)
    ...:     

In [16]: r3()
0.7356688330037287

In [17]: def r4(n=10000000):
    ...:     st = perf_counter()
    ...:     for _ in range(n):
    ...:         a.__class__.b
    ...:     print(perf_counter() - st)
    ...:     

In [18]: r4()
0.4558341469964944
```